fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraSearchIT.java

```diff
@@ -124,11 +136,184 @@ public void testWildcardMatchingOnPartialFedoraId() throws Exception {
+            assertEquals(NO_CONTENT.getStatusCode(), getStatus(response));
+        }
+
+        try (final CloseableHttpResponse response = execute(new HttpGet(searchUrl))) {
+            assertEquals(OK.getStatusCode(), getStatus(response));
+            final ObjectMapper objectMapper = new ObjectMapper();
+            final SearchResult result = objectMapper.readValue(response.getEntity().getContent(),
+                    SearchResult.class);
+            assertEquals("expected 0 items where condition = " + condition, 0,
+                    result.getItems().size());
+        }
+    }
+
+    public void testModifiedGreaterAndLessThan() throws Exception {
+        final var id = getRandomUniqueId();
+        final int count = 1;
+        final var instant = Instant.now();
+        final var now = instant.toString();
+        final var tomorrow = instant.plus(Duration.ofDays(1)).toString();
+        createResources(id, count);
+        final var fedoraId = FEDORA_ID_PREFIX + "/" + id;
+        final var fedoraIdCondition = FEDORA_ID + "=" + fedoraId;
+        final var lessThanNow = MODIFIED + "<" + now;
+        final var greaterThanNow = MODIFIED + ">" + now;
+        final var lessThanTomorrow = MODIFIED + "<" + tomorrow;
+        final var greaterThanTomorrow = MODIFIED + ">" + tomorrow;
+
+        //no results for resources modified before now
+        String searchUrl =
+                getSearchEndpoint() + "condition=" + encode(fedoraIdCondition) + "&" + lessThanNow;
+        try (final CloseableHttpResponse response = execute(new HttpGet(searchUrl))) {
+            assertEquals(OK.getStatusCode(), getStatus(response));
+            final ObjectMapper objectMapper = new ObjectMapper();
+            final SearchResult result = objectMapper.readValue(response.getEntity().getContent(),
+                    SearchResult.class);
+            assertEquals("expected no results", 0,
+                    result.getItems().size());
+            result.getItems().stream().map(x -> x.get("fedora_id"))
+                    .forEach(x -> assertTrue(x.toString().equals(fedoraId)));
+        }
+        //no results for resources modified after tomorrow
+        searchUrl =
+                getSearchEndpoint() + "condition=" + encode(fedoraIdCondition) + "&" + greaterThanTomorrow;
+        try (final CloseableHttpResponse response = execute(new HttpGet(searchUrl))) {
+            assertEquals(OK.getStatusCode(), getStatus(response));
+            final ObjectMapper objectMapper = new ObjectMapper();
+            final SearchResult result = objectMapper.readValue(response.getEntity().getContent(),
+                    SearchResult.class);
+            assertEquals("expected no results", 0,
+                    result.getItems().size());
+            result.getItems().stream().map(x -> x.get("fedora_id"))
+                    .forEach(x -> assertTrue(x.toString().equals(fedoraId)));
```

Comment:
The size of `result.getItems().size()` is asserted as 0 above, so this stream is never run and can probably be removed.

Reply:
Good eye.